### PR TITLE
Add support for kubernetes identity file format

### LINKED ISSUE #2825: Implement 'tctl auth sign --type=kubernetes'
**What happened**: Trying to run `tsh login -i identity_file.pem --proxy teleport.example.com` results in `error: -i flag cannot be used here` - this means that a long-lived certificate (such as those generated for use with automation as per https://gravitational.com/teleport/docs/user-manual/#ssh-certificates-for-automation) can't log into a cluster to get `~/.kube/config` updated with Kubernetes credentials.

**What you expected to happen**: To have a method available to authenticate with the Kubernetes apiserver as well as SSH.

The recommendation from @klizhentas is to implement `tctl auth export --kubeconfig` which will generate a kubeconfig for the specified user that can be provided to `kubectl` and used to run operations on the Kubernetes cluster.

**How to reproduce it (as minimally and precisely as possible)**: `tsh login -i identity_file.pem --proxy teleport.example.com`

**Environment**:
- Teleport version (use `teleport version`): `Teleport Enterprise v4.0.0git:v4.0.0-0-gc7f55ac3 go1.12.1`
- Tsh version (use `tsh version`): `Teleport v3.2.6 git:v3.2.6-0-g67b4ddfb go1.11.5`
- OS (e.g. from /etc/os-release): `Fedora 29`

### PR DESCRIPTION
**Pull request by awly**
There are two new ways you can generate a kubeconfig:
- `tctl auth sign --user=foo --format=kubernetes --out=kubeconfig` for
  admins
- `tsh login --format=kubernetes -o kubeconfig` for users

This allows admins to generate long-lived kubeconfigs for e.g. CI
systems.

A tricky part is getting the kubernetes endpoint for a proxy in `tctl`.
It does its best to guess the address, but falls back to asking user to
pass `--proxy` flag.
It looks like right now, the proxy info available via the auth server's
API doesn't have kubernetes public_addr for proxies.

Fixes #2825

### CODE REVIEW

**Comment by awly** [comment]
Ping for review.

**Review by webvictim — APPROVED** [review]
LGTM. I would wait for approval from one of the others though.

**Review by russjones — APPROVED** [review]
Missing test coverage for Kubernetes format in `TestWrite`.

**Pull request merged**
